src/services/ProductsService.ts

```diff
@@ -0,0 +1,117 @@
+import fs from "fs-extra"
+import Product from "../types/Product";
+import CustomError from "../types/CustomError";
+import path from "path"
+import StatusCode from "../types/StatusCode";
+
+const filePath = path.resolve(__dirname, '..', 'products.json');
+
+export const getAllProducts = async (): Promise<Product[]> => {
+	try {
+		const packageObj = await fs.readJson(filePath);
+		const products = packageObj.products.filter((product: Product) => !product.deleted)
+
+		if (!products || !products.length) {
+			throw new CustomError("Not found products", StatusCode.NOT_FOUND)
+		}
+
+		return products;
+	} catch (err) {
+		console.error(err);
+		throw new CustomError("Unable to receive the products", StatusCode.INTERNAL_SERVER_ERROR)
+	}
+};
+
+export const getProductById = async (id: string): Promise<Product> => {
+	try {
+		const packageObj = await fs.readJson("src/products.json");
+		const product = packageObj.products.find((p: Product) => p.id === id && !p.deleted);
```

Comment:
you are using this part of code 
`	const product = packageObj.products.find((p: Product) => p.id === id && !p.deleted);

		if (!product) {
			throw new CustomError("Not found products", StatusCode.NOT_FOUND)
		}`
		several time, you may write a middleware, and add in the routes you need, to avoid code redundancy